### PR TITLE
Fix tests

### LINKED ISSUE #18299: Test failures: TestDateTime.FileTimeOperators and TestDateTime.SetFromUTCDateTime
### Describe the bug
Here is a clear and concise description of what the problem is:

These two tests are failing for Kodi 19a1:
TestDateTime.FileTimeOperators
TestDateTime.SetFromUTCDateTime

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Tests should pass :)

## Actual Behavior

The tests fail. Here are some excerpts from LastTest.log:
```
444/547 Testing: TestDateTime.FileTimeOperators
444/547 Test: TestDateTime.FileTimeOperators
Command: "/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/kodi-19.0_alpha1_build/kodi-test" "--gtest_filter=TestDateTime.FileTimeOperators"
Directory: /var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/kodi-19.0_alpha1_build
"TestDateTime.FileTimeOperators" start time: Aug 17 06:48 CEST
Output:
----------------------------------------------------------
Note: Google Test filter = TestDateTime.FileTimeOperators
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from TestDateTime
[ RUN      ] TestDateTime.FileTimeOperators
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:43: Failure
Value of: dateTime1 < fileTime
  Actual: false
Expected: true
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:44: Failure
Value of: dateTime1 > fileTime
  Actual: true
Expected: false
[  FAILED  ] TestDateTime.FileTimeOperators (0 ms)
[----------] 1 test from TestDateTime (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (49 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestDateTime.FileTimeOperators

 1 FAILED TEST
Clean shutdown of TestGlobalPattern1
<end of output>
```
```
458/547 Testing: TestDateTime.SetFromUTCDateTime
458/547 Test: TestDateTime.SetFromUTCDateTime
Command: "/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/kodi-19.0_alpha1_build/kodi-test" "--gtest_filter=TestDateTime.SetFromUTCDateTime"
Directory: /var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/kodi-19.0_alpha1_build
"TestDateTime.SetFromUTCDateTime" start time: Aug 17 06:48 CEST
Output:
----------------------------------------------------------
Note: Google Test filter = TestDateTime.SetFromUTCDateTime
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from TestDateTime
[ RUN      ] TestDateTime.SetFromUTCDateTime
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:250: Failure
Expected equality of these values:
  dateTime3.GetYear()
    Which is: 1922
  1991
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:251: Failure
Expected equality of these values:
  dateTime3.GetMonth()
    Which is: 11
  5
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:253: Failure
Expected equality of these values:
  dateTime3.GetHour()
    Which is: 19
  12
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:254: Failure
Expected equality of these values:
  dateTime3.GetMinute()
    Which is: 36
  34
/var/tmp/portage/media-tv/kodi-19.0_alpha1-r1/work/xbmc-19.0a1-Matrix/xbmc/test/TestDateTime.cpp:255: Failure
Expected equality of these values:
  dateTime3.GetSecond()
    Which is: 26
  56
[  FAILED  ] TestDateTime.SetFromUTCDateTime (0 ms)
[----------] 1 test from TestDateTime (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (48 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestDateTime.SetFromUTCDateTime

 1 FAILED TEST
Clean shutdown of TestGlobalPattern1
<end of output>
```
Full `LastLog.log`: https://737544.bugs.gentoo.org/attachment.cgi?id=655024
build output: https://bugs.gentoo.org/attachment.cgi?id=655018

### To Reproduce

1. Run the Kodi 19a1 tests

### Debuglog

The debuglog can be found here:
n/a

### Screenshots 
Here are some links or screenshots to help explain the problem:

n/a

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

Originally reported at https://bugs.gentoo.org/737544

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Gentoo/Linux
 - Kodi version: 19a1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by basilgello**
## Description

Fixes #18299 on Linux and amends #18011 

## Motivation and Context

Building Debian package requires full testsuite to pass unless "-notest" CLI switch is given.
However, official builds include testsuite runs and having even one failing test results in
a whole failed build.

## How Has This Been Tested?

Built Debian package on Debian sid / unstable.

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by basilgello** [comment]
@phunkyfish @lrusak

File: xbmc/platform/linux/test/TestSysfsPath.cpp [hunk 1]
```diff
@@ -19,19 +19,17 @@ struct TestSysfsPath : public ::testing::Test
...
     const char *test_tmpdir = getenv("TMPDIR");
     if (test_tmpdir && test_tmpdir[0] != '\0') {
       tmpdir.assign(test_tmpdir);
     }
 
     std::stringstream ss;
-    unsigned long long timestamp = duration_cast< nanoseconds >(
-      system_clock::now().time_since_epoch()).count();
+    const std::string filename =
+      KODI::UTILS::RandomAlphaNumString(12);
```
Comment (phunkyfish) [inline xbmc/platform/linux/test/TestSysfsPath.cpp:30]: Why two lines?

File: xbmc/utils/Random.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#include <random>
+#include <string>
+#include <vector>
+
+namespace KODI
+{
+namespace UTILS
+{
+
+char_array alnum_charset()
```
Comment (phunkyfish) [inline xbmc/utils/Random.cpp:21]: `AlnumCharset`

File: xbmc/utils/Random.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+{
+    return char_array(
```
Comment (phunkyfish) [inline xbmc/utils/Random.cpp:23]: `charArray`

File: xbmc/utils/Random.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	});
+};
+
+std::string RandomAlphaNumString(int length)
+{
+  const auto charset = alnum_charset();
+
+  std::random_device rd;
+  std::mt19937 mt(rd());
+  std::uniform_int_distribution<> dist(0, charset.size()-1);
```
Comment (phunkyfish) [inline xbmc/utils/Random.cpp:45]: Space before and after `-1`

File: xbmc/utils/Random.h [hunk 5]
```diff
@@ -15,12 +15,18 @@ namespace KODI
 {
 namespace UTILS
 {
+
+typedef std::vector<char> char_array;
```
Comment (phunkyfish) [inline xbmc/utils/Random.h:19]: Do we really need a typedef here?

**Review by phunkyfish — CHANGES_REQUESTED** [review]
It would make sense to run git-clang-format on your changes. You may want to switch clang-format off /on for the char array.

**Comment by lrusak** [comment]
I feel like adding a whole new class just for tests is a bit overkill. Other changes are probably fine. This will all be wiped out hopefully when I finish the work to use std::chron/std::date

**Comment by basilgello** [comment]
> I feel like adding a whole new class just for tests is a bit overkill. Other changes are probably fine. This will all be wiped out hopefully when I finish the work to use std::chrono/std::date

It is not a class but a separate file to make ld.gold happy about "duplicate definition" at link-time.
I can make this function a part of TestSysfsPath.cpp and leave Random.h intact but I thought we
might need a standard random string generator as part of KODI::UTILS. Do we need it?

I want to know this before resolving issues @phunkyfish mentioned.

**Comment by phunkyfish** [comment]
I think make it local to the test currently as that is the only place it is used. If we need to generalise it later as we have another use case we can do that then.

**Comment by lrusak** [comment]
you could always just use the uuid generator instead of implementing something new. See, https://github.com/xbmc/xbmc/blob/master/xbmc/utils/StringUtils.h#L312

**Comment by basilgello** [comment]
Makes sense, thanks! Fixed!

**Review by phunkyfish — APPROVED** [review]

**Review by lrusak — APPROVED** [review]
thanks for the great work! LGTM

**Pull request merged**